File: 1176-diet-plan-performance/1176-diet-plan-performance.cpp
class Solution {
public:
    int dietPlanPerformance(vector<int>& calories, int k, int lower, int upper) {
        int total = 0;
        int points = 0;
        
        for(int i = 0; i < k; i++)
            total += calories[i];
        if( total < lower )
            points--;
        else if ( total > upper )
            points++;
        
        for( int i = k; i < calories.size(); i++ ) {
            total += calories[i] - calories[i-k];
        
        if( total < lower )
            points--;
        else if ( total > upper )
            points++;
        }
        
        return points;
    }
};

File: 1176-diet-plan-performance/1176-diet-plan-performance.py
class Solution(object):
    def dietPlanPerformance(self, calories, k, lower, upper):
        """
        :type calories: List[int]
        :type k: int
        :type lower: int
        :type upper: int
        :rtype: int
        """
        points = 0
        winSum = sum(calories[:k])
        
        if winSum < lower:
            points -= 1
        elif winSum > upper:
            points += 1
        
        for i in range(k, len(calories)):
            winSum = winSum + calories[i] - calories[i-k]
            if winSum < lower:
                points -= 1
            elif winSum > upper:
                points += 1
                
        return points